src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -492,6 +492,51 @@ export default class DefaultAudioVideoController
     );
   }
 
+  async replaceLocalVideo(callback: () => void): Promise<void> {
+    let videoStream: MediaStream | null = null;
+    try {
+      videoStream = await this.mediaStreamBroker.acquireVideoInputStream();
+    } catch (error) {
+      this.logger.info('could not acquire video stream from mediaStreamBroker');
+    }
+    if (!videoStream || videoStream.getVideoTracks().length < 1) {
+      throw new Error('could not acquire video track');
+    }
+
+    const videoTrack = videoStream.getVideoTracks()[0];
+    if (!this.meetingSessionContext || !this.meetingSessionContext.peer) {
+      throw new Error('no active meeting and peer connection');
+    }
```

Comment:
Since there is no peer connection, replacing video on peer connection is a not permitted.
If users use default device controller and play with VideoTransformDevice, `replaceLocalVideo` is  guarded by `hasLocalTileStarted`. If users wants to implement their own default device controller, call `replaceLocalVideo` is not permitted.

Reply:
How can I preview my emoji/face filter in a lobby view?

Reply:
lobby view does not depend on `replaceLocalVideo` (same as audio).